File: app/src/main/java/lk/cn/com/newbanner/ItemAdapter.java
package lk.cn.com.newbanner;

import android.content.Intent;
import android.view.View;

import com.chad.library.adapter.base.BaseQuickAdapter;
import com.chad.library.adapter.base.BaseViewHolder;

public class ItemAdapter extends BaseQuickAdapter<ActivityItem, BaseViewHolder> {

        public ItemAdapter() {
            super(R.layout.item_main);
        }

        @Override
        protected void convert(BaseViewHolder helper, final ActivityItem item) {
            if (item != null) {
                helper.setText(R.id.btn, item.name);
                helper.getView(R.id.btn).setOnClickListener(new View.OnClickListener() {
                    @Override
                    public void onClick(View v) {
                        Intent i = new Intent(mContext, item.cls);
                        mContext.startActivity(i);
                    }
                });
            }
        }
    }

File: app/src/main/java/lk/cn/com/newbanner/recycle/ItemAdapter.kt
package lk.cn.com.newbanner.recycle

import android.widget.TextView
import com.chad.library.adapter.base.BaseQuickAdapter
import com.chad.library.adapter.base.BaseViewHolder
import lk.cn.com.newbanner.R
import java.util.*




/**
 * 卡券Item
 */
public class ItemAdapter(data: ArrayList<Item>) : BaseQuickAdapter<Item, BaseViewHolder>(R.layout
        .item_recycle, data), IItemTouchHelper {

    override fun convert(helper: BaseViewHolder?, item: Item?) {
        var tv = helper?.getView<TextView>(R.id.tv)
        tv?.setBackgroundColor(item?.color ?: 0)
        tv?.text = item?.txt
        if (tv?.height != item?.height)
        {
            var lp = tv?.layoutParams
            lp?.height = item?.height
        }

//        helper?.addOnClickListener(R.id.tv)
    }

    override fun onItemMove(fromPosition: Int, toPosition: Int) {
        if (fromPosition >= 0 && toPosition >= 0 && fromPosition != toPosition &&
                fromPosition < data.size && toPosition < data.size) {

            if (fromPosition < toPosition) {
                //分别把中间所有的 item 的位置重新交换
                for (i in fromPosition until toPosition) {
                    Collections.swap(data, i, i + 1)
                }
            } else {
                for (i in fromPosition downTo toPosition + 1) {
                    Collections.swap(data, i, i - 1)
                }
            }

            notifyItemMoved(fromPosition, toPosition)
        }
    }

    override fun onItemDismiss(position: Int) {
        if (position > -1 && position < data.size) {
            data.removeAt(position)
            notifyItemRemoved(position)
        }
    }
}